osu.Game/OsuGame.cs

```diff
@@ -276,12 +278,27 @@ protected override void LoadComplete()
                 switch (settings.State)
                 {
                     case Visibility.Hidden:
-                        intro.MoveToX(0, SettingsOverlay.TRANSITION_LENGTH, Easing.OutQuint);
+                        backgroundOffset -= ToolbarButton.WIDTH / 2;
+                        break;
+                    case Visibility.Visible:
+                        backgroundOffset += ToolbarButton.WIDTH / 2;
+                        break;
+                }
+                intro.MoveToX(backgroundOffset, SettingsOverlay.TRANSITION_LENGTH, Easing.OutQuint);
+            };
+
+            notificationOverlay.StateChanged += delegate
+            {
+                switch (notificationOverlay.State)
+                {
+                    case Visibility.Hidden:
+                        backgroundOffset += ToolbarButton.WIDTH / 2;
                         break;
                     case Visibility.Visible:
-                        intro.MoveToX(SettingsOverlay.SIDEBAR_WIDTH / 2, SettingsOverlay.TRANSITION_LENGTH, Easing.OutQuint);
+                        backgroundOffset -= ToolbarButton.WIDTH / 2;
                         break;
                 }
+                intro.MoveToX(backgroundOffset, NotificationOverlay.TRANSITION_LENGTH, Easing.OutQuint);

```

Comment:
this is really bad code duplication. just make an inline function and bind the same thing to both `StateChanged` events.

```
void overlayStateChanged()
{
    float offset = 0;
    if (notificationOverlay.State == Visible) offset -= width /2;
    if (settingsOverlay == Visible) offset += width /2;
}
```